repository/instance_repository.go

```diff
@@ -112,6 +114,50 @@ func (repo *InstanceRepository) AddDimensions(i *model.Instance) error {
 	return nil
 }
 
+// CreateCodeRelationship links an instance to a code for the given dimension option
+func (repo *InstanceRepository) CreateCodeRelationship(i *model.Instance, code string) error {
+
+	if i == nil {
+		return errors.New("instance is required but was nil")
+	}
+	if len(i.InstanceID) == 0 {
+		return errors.New("instance id is required but was empty")
+	}
+	if len(code) == 0 {
+		return errors.New("code is required but was empty")
+	}
+
+	instanceLabel := fmt.Sprintf(instanceLabelFmt, i.GetID())
+	stmt := fmt.Sprintf(createInstanceToCodeRelStmt, instanceLabel)
+	params := map[string]interface{}{"code": code}
+
+	logDebug := map[string]interface{}{
+		"statement":   stmt,
+		"params":      params,
+		"instance_id": i.InstanceID,
+		"code":        code,
+	}
+	loggerI.Info("executing create code relationship statement", logDebug)
+
+	result, err := repo.neo4j.ExecStmt(repo.conn, stmt, params)
+	if err != nil {
+		return errors.Wrap(err, "neo4j.ExecStmt returned an error")
+	}
+
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return errors.Wrap(err, "result.RowsAffected() returned an error")
+	}
+
+	logDebug["rows_affected"] = rowsAffected
+	if rowsAffected != 1 {
+		return errors.New("failed to find the code node to link to the instance node")
+	}
+
+	loggerI.Info("create code relationship success", logDebug)
```

Comment:
Why are we using this loggerI? It is very strange to create a wrapper to the log package in go